### PR TITLE
DEPR: Removing previously deprecated Categorical.labels (#6581)

### PR DESCRIPTION
**Pull request by datapythonista**
- [X] related to #6581
- [X] tests added / passed
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [X] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v0.23.0.txt [hunk 1]
```diff
@@ -314,6 +314,7 @@ Removal of prior version deprecations/changes
 - The ``Panel4D`` and ``PanelND`` classes have been removed (:issue:`13776`)
 - The ``Panel``class has dropped the ``to_long``and ``toLong`` methods (:issue:`19077`)
 - The options ``display.line_with`` and ``display.height`` are removed in favor of ``display.width`` and ``display.max_rows`` respectively (:issue:`4391`, :issue:`19107`)
+- The ``labels`` attribute of the ``Categorical`` class has been drop in favor of ``codes``
```
Comment (jreback) [inline doc/source/whatsnew/v0.23.0.txt]: drop -> removed

``Categorical.labels`` in favor of  ``:attribute`Categorical.codes` ``

add the issue number, the original if you can dig it up and this one

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/whatsnew/v0.23.0.txt (hunk 1 above)
Comment (datapythonista) [reply doc/source/whatsnew/v0.23.0.txt]: Thanks for the comments! There is not issue related to deprecating or removing `labels`. The commits didn't reference any, and searching in the issue tracker didn't find anything either. I guess by "this one" you mean #6581, right? I updated the PR with it.

File: doc/source/whatsnew/v0.23.0.txt [hunk 2]
```diff
@@ -314,6 +314,7 @@ Removal of prior version deprecations/changes
 - The ``Panel4D`` and ``PanelND`` classes have been removed (:issue:`13776`)
 - The ``Panel``class has dropped the ``to_long``and ``toLong`` methods (:issue:`19077`)
 - The options ``display.line_with`` and ``display.height`` are removed in favor of ``display.width`` and ``display.max_rows`` respectively (:issue:`4391`, :issue:`19107`)
+- The ``labels`` attribute of the ``Categorical`` class has been removed in favor of attribute:`Categorical.codes` (:issue:`6581`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.23.0.txt]: use: ``:attribute:`Categorical.codes` ``

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/whatsnew/v0.23.0.txt (hunk 1 above)
Comment (jreback) [reply doc/source/whatsnew/v0.23.0.txt]: no I dont mean #6581 I mean the issue / PR where it was originally deprecated. Yes its a commit hash, but more useful is a specific PR.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks!

**Pull request merged**